undertow/src/main/java/org/wildfly/extension/undertow/UndertowSubsystemAdd.java

```diff
@@ -102,8 +105,9 @@ protected void performBoottime(OperationContext context, ModelNode operation, Re
         final String defaultSecurityDomain = UndertowRootDefinition.DEFAULT_SECURITY_DOMAIN.resolveModelAttribute(context, model).asString();
 
         final ModelNode instanceIdModel = UndertowRootDefinition.INSTANCE_ID.resolveModelAttribute(context, model);
-        final String instanceId = instanceIdModel.isDefined() ? instanceIdModel.asString() : null;
-
+        final ModelNode uuidInstanceIdModel = UndertowRootDefinition.UUID_INSTANCE_ID.resolveModelAttribute(context, model);
+        final String instanceId = instanceIdModel.isDefined() ? (uuidInstanceIdModel.asBoolean() ?
```

Comment:
IMO, this logic actually belongs in org.wildfly.extension.undertow.Server#getRoute().

The getRoute() method should obfuscate the entire route (which may also include the server name) when the obfuscation flag is enabled.